### ISSUE TITLE
To all Iris lovers

### ISSUE THREAD

**Issue by speedwheel** [kataras/iris#796]
Dear fellow gophers, Iris users and lurkers, we are all lucky to be here and use this wonderful go framework called Iris.

 I am glad to see that Iris has grown into a mature software that is used and trusted by many of us in our projects, from small individual developers to large teams.

There have been a lot of good changes lately that made Iris more powerful than ever before (and this is just the start), easy to use, but in the same time and with very good performance.

But in my opinion the most important thing what makes Iris truly different from other frameworks, is its documentation, which is very important for the our user base, especially for newcomers. And Iris docs are doing a tremendous job explaining all the aspects of the framework with it's documentation by using 
 a big variety of examples.

Now having all that said, let's move to the "issue" part for this ticket.
I already mentioned that the documentations are plain awesome, but only if you understand english!
Now let's point out the second most used spoken language by Golang devs, well I think you already know of whom I am talking about... you probably already guessed, it's the Chinese users!

I have also noticed myself that more and more Chinese users are opening different issues here and probably many of them looking around trying to grasp and get a hold of using Iris, but unfortunately the language barrier stops them from achieving that, since we all know that a very small percentage of China's population actually speaks english.

So by all means I am making a call to all Chinese Iris users here that know English and want to help and make Iris a more welcoming place for there fellow compatriots by translating as much as they can of the Iris project.

Now let's get down to business and see what exactly I mean by that. These are the main things that need to be translated, ordered by priority:

1.  https://github.com/kataras/iris - there are two .md files that need to be translated, README.md and README_NEXT.md, please submit a PR for this with the following file naming: README_CN.md and README_NEXT_CN.md;

2. https://github.com/kataras/iris/tree/master/_examples - there is one .md file here that needs to be translated, README.md, please submit a PR for this with the following file naming:  README_CN.md;

3. https://iris-go.com/ - here I decided to make a new website using the .cn TLD, so the final domain name will be "iris-go.cn", where we will be adding the same information from iris-go.com, but translated in Chinese.

Also a good thing to keep in mind if you want take part to this, is to check also the English version for mistakes doing the translation process.

Let's all build a better web together, that's what open-source it's all about!

![download](https://user-images.githubusercontent.com/5404341/31921416-fc6dcb4c-b877-11e7-8906-fb03502867f3.jpg)

**Comment by unreadable** [comment]
There's more english speaker in China than USA...I don't know where you found that small percentage oO. Either way, PRs are wellcome ;)

**Comment by speedwheel** [comment]
@unreadable https://en.wikipedia.org/wiki/List_of_countries_by_English-speaking_population

**Comment by a233894432** [comment]
在中国,用你这个框架的人还是比较多的.
我在好几个论坛上面都有介绍的.

**Comment by Zeno-Code** [comment]
I think Beego's STAR far more than IRIS, in addition to release time earlier, another reason is that a large number of Chinese users

I have a question
I need 100% according to the original translation, or can delete some unnecessary statements according to the Chinese situation

My English need to use Google translate, please forgive me

But I've been relatively familiar with IRIS and I can translate relatively authentic Chinese

**Comment by kataras** [comment]
Yes @Zeno-Code, Iris is just 1 and a half years, beego is 5 years I think and I don't know how many authors beego has but Iris has only one... I'm doing my bests and sometimes I have help from people like you, especially on bug reports, translations and english grammar fixes and I am very thanful for this.

About the translation, yes I'll add these links for HISTORY_ZH.md (you've just created) and README_ZH.md (you've just updated) to the english README.md and HISTORY_ZH.md files as well, just tell me when you finish them, and yes you can remove any unnecessary parts from the original file, do it as much naturally as you want, it's totally fine with me!

**Comment by kataras** [comment]
Yes @Zeno-Code, Iris is just 1 and a half years, beego is 5 years I think and I don't know how many authors beego has but Iris has only one... I'm doing my bests and sometimes I have help from people like you, especially on bug reports, translations and english grammar fixes and I am really very thanful for this.

About the translation, yes I'll add these links for HISTORY.md (you've just created) and README_ZH.md (you've just updated) to the english README.md and HISTORY_ZH.md files as well, just tell me when you finish them, and yes you can remove any unnecessary parts from the original file, do it as much naturally as you want, it's totally fine with me!

**Comment by TechMaster** [comment]
Hi Kataras,

We are going to create a online training course how to develop web application using Iris Framework. Language will be Vietnamese.

The course will be released partly in Feb 2018 and completed in April 2018.

**Comment by kataras** [comment]
Hi @TechMaster,

That is awesome! The whole community will be glad to help you, including me, if you need any help while creating the course.

**Comment by kataras** [comment]
New Translations for README are required, the content has drammatically changed(got minified as the msot content now lives inside its [iris/wiki](https://github.com/kataras/iris/wiki)) as of Iris 11.2. Thanks for the support!

**Comment by handuy** [comment]
Hi Kataras,
I will open a PR to translate the ReadMe of these to Vietnamese:
- https://github.com/kataras/iris/tree/master/_examples
- https://github.com/kataras/iris

But I also want to contribute translating the IRIS wiki:
https://github.com/kataras/iris/wiki
Can you tell me how to do it ? Thanks :D

**Comment by kataras** [comment]
Hello @handuy, sorry for the delay. I  did reply to the e-mail you sent me a week ago as well.

Let me put it here for anyone who is looking to translate the wikis too.


Github Wikis cannot render branches except  `master` but you can clone it to a repository of yours and push the translation there, here is how.

Iris Wiki can be cloned, like any github repository:

```sh
$ git clone https://github.com/kataras/iris/wiki.git
```

1.  So, create a new github repository, e.g. "github.com/$your_username/iris.vi" (where "vi" is the two-letter language code of Vietnamese). Note that this will be an empty repository, we only need its "/wiki",
2.  Enable Wikis in the github repository settings (if not enabled by-default),
3. Create a new folder, i.e "iris.vi.wiki", and open a terminal/shell window,
4. Clone the empty wiki locally `git clone https://github.com/$your_username/iris.vi/wiki.git .` ,
5. Navigate somewhere else, grab the Iris' wikis content by `git clone https://github.com/kataras/iris/wiki.git` and copy-paste all files to your  "iris.vi.wiki" folder,
6. Make your changes and do a git push like any github repository,
7. When you finish the translation just send me a message or open a new github iris issue to share your link, I will add a link on the kataras/iris/wiki too.


Looking forward for the Vietnamese READMEs and wiki!!

I also posted the Video Course about Go and Iris you guys did on the ["Thank you" page of the request iris book form](https://iris-go.com/forms/iris/thank_you/).

> https://techmaster.vn/khoa-hoc/25617/xay-dung-ung-dung-web-ban-hang-bang-golang

Thanks,
Gerasimos Maropoulos

**Comment by 1hitOVER** [comment]
Hello @kataras,

I can help with Spanish translation. :D

**Comment by kataras** [comment]
Hello @1hitOVER,

That's great news! Send me a message, if you need my assistance on anything. Wikis? new docs.iris-go.com subdomain for spanish? Anything.

**Comment by 1hitOVER** [comment]
Hi @kataras,

I just opened PR for spanish translation of readme and history files.

I'll work on translation of _examples/README file.

PR #1344

**Comment by jinil-ha** [comment]
Hello @kataras 
I think I can translate to Korean and Japanese.
May I?

**Comment by kataras** [comment]
Thank you @1hitOVER  @jinil-ha, PRs accepted, good work! Thank you for your contribution to the community!

**Comment by 1hitOVER** [comment]
Hi @kataras ,

I updated spanish translations for v12 in this PR
#1387

**Comment by kataras** [comment]
@koddr added Russian translations with [PR#1412](https://github.com/kataras/iris/pull/1412). Thanks!

Rel to: https://github.com/kataras/iris/issues/1411

**Comment by Ynniss** [comment]
thinking of working on the french translation, even if it's not on the todo list.
Is that good for you ?

**Comment by kataras** [comment]
Sure @Ynniss!!

**Comment by kataras** [comment]
French translation added by @Natnatf, PR: https://github.com/kataras/iris/pull/1501

**Comment by fengahan** [comment]
www.studyiris.com 中文社区成立,提供完善的文档翻译。

**Comment by ahliweb** [comment]
Has anyone started translating into Indonesian?

**Comment by joewongex** [comment]
> [www.studyiris.com](http://www.studyiris.com) 中文社区成立,提供完善的文档翻译。
不能访问了？

**Comment by kataras** [comment]
Hello @ahliweb , there is Indonesian [video course on youtube](https://www.youtube.com/watch?v=uXiNYhJqh2I&index=1&list=PLMrwI6jIZn-1tzskocnh1pptKhVmWdcbS)  with 23 parts  covering the MVC features of Iris, however you must update the path and read the latest [changelog](https://github.com/kataras/iris/blob/master/HISTORY.md). There are no documentation translations in Indonesian, as far as I am aware of. If you are going to do it please do it by cloning [that repository](https://github.com/kataras/build-a-better-web-together) instead, it's a live update of the new docs.

**Comment by kataras** [comment]
@joewongex There is Chinese Iris wiki for v12.1.8 written by the amazing @snowlyg at: https://github.com/snowlyg/iris/wiki. And also I am preparing a new website for documentation translation and editing that will be available on v12.2.0 release (which requires new documentation).
